lib/collection/src/config.rs

```diff
@@ -146,6 +146,16 @@ impl CollectionConfig {
             validation::warn_validation_errors("Collection configuration file", errs);
         }
     }
+
+    pub fn get_quantization_named_configs(&self) -> BTreeMap<String, Option<QuantizationConfig>> {
```

Comment:
In `config.get_all_vector_params()` we already assign a quantization configuration to the data  for every config:

https://github.com/qdrant/qdrant/blob/948b630aa7c8be919bbb065e7bd5add452c07e61/lib/collection/src/config.rs#L185-L189

In other words: the data for every vector returned here has its specific quantization config, or the global config if it was not set, or no config if none was set anywhere.

This means that, the data returned here, always has the correct quantization configuration on every vector. I think it may be good to use this here as well, instead of manually grabbing these configurations.

In fact, we should probably remove the _global_ quantization configuration in all deeper structures, and just use vector specific configurations. That would minimize confusion, since we probably only need the vector specific ones internally.